Tropos/Models/DailyForecast.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+@objc class DailyForecast: NSObject {
+    let date: NSDate
+    let conditionsDescription: String
+    let highTemperature: Temperature
+    let lowTemperature: Temperature
+    
+    init(JSON: Dictionary<String, AnyObject>) {
```

Comment:
I'll update this topic and show how I see implementation of what we already talked about.

`DailyForecast`

	@objc class DailyForecast: NSObject {
	    let date: NSDate
	    let conditionsDescription: String
	    let highTemperature: Temperature
	    let lowTemperature: Temperature
	    
	    init(time: NSTimeInterval, conditionsDescription: String, highTemperature: Int, lowTemperature: Int) {
	        self.date = NSDate(timeIntervalSince1970: time)
	        self.conditionsDescription = conditionsDescription
	        self.highTemperature = Temperature(fahrenheitValue: highTemperature)
	        self.lowTemperature = Temperature(fahrenheitValue: lowTemperature)
	    }
	    
	    class func dailyForecastWithJSON(JSON: Dictionary<String, AnyObject>) -> DailyForecast? {
	        if let time = JSON["time"] as? NSTimeInterval,
	            let conditionsDesc = JSON["icon"] as? String,
	            let tempMax = JSON["temperatureMax"] as? Int,
	            let tempMin = JSON["temperatureMin"] as? Int {
	            return DailyForecast(time: time, conditionsDescription: conditionsDesc, highTemperature: tempMax, lowTemperature: tempMin)
	        }
	        return nil
	    }
	}



`TRWeatherUpdate` - enumerating through forecasts:

    NSMutableArray *dailyForecasts = [NSMutableArray array];
    for (NSUInteger index = 1; index < 4; index++) {
        DailyForecast *dailyForecast = [DailyForecast dailyForecastWithJSON:currentConditionsJSON[@"daily"][@"data"][index]];
        if (dailyForecast != nil) {
            [dailyForecasts addObject:dailyForecast];
        }
    }



`TRWeatherViewController` - updating forecasts views:

    [self.controller.dailyForecastViewModels subscribeNext:^(NSArray *viewModels) {
        NSUInteger index = 0;
        for (TRDailyForecastView *view in forecastViews) {
            TRDailyForecastViewModel *model = index < viewModels.count ? viewModels[index] : nil;
            if (model != nil) {
                view.hidden = NO;
                view.viewModel = model;
            } else {
                view.hidden = YES;
            }
            index++;
        }
    }];

This allows us to hide forecasts that has not been created because of some backend issues. Behaviour the same like proposed by @jakecraige  - https://cloud.githubusercontent.com/assets/3799709/8501575/b69945b6-215b-11e5-8ba3-01f825992b2b.png